### PR TITLE
board: arm: Add support for WeAct Mini STM32H7B0

### PR DESCRIPTION
**Pull request by ric96**
This board is mostly the same as mini_stm32h743.

Noatable diffrences for dt:
- rcc config
- octospi in quadspi mode
- pinmux naming for USB and quadspi

https://builds.zephyrproject.io/zephyr/pr/80401/docs/boards/weact/mini_stm32h7b0/doc/index.html

### CODE REVIEW

File: boards/weact/mini_stm32h7b0/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+More information about STM32H7BO can be found here:
+
+- `STM32H7B0VB on www.st.com`_
+- `STM32H7B0VB reference manual`_
+- `STM32H7B0VB datasheet`_
+
+Supported Features
+==================
+
+The mini_stm32h7b0 board configuration supports the following hardware features:
```
Comment (nordicjm) [inline boards/weact/mini_stm32h7b0/doc/index.rst:82]: *board target, and enclose board name with \`\`

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/weact/mini_stm32h7b0/doc/index.rst (hunk 1 above)
Comment (ric96) [reply boards/weact/mini_stm32h7b0/doc/index.rst:82]: @nordicjm currently it is consistent with ever other ST and WeAct boards doc. I can change it if its required but it breaks consistency. What I will do atm is make it on par with a43009079f6670ef1085603b0474d147ffdb76f4

Comment (nordicjm) [reply boards/weact/mini_stm32h7b0/doc/index.rst:79]: board configuration does not exist and is invalid, board target is correct, see https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-terminology

Comment (ric96) [reply boards/weact/mini_stm32h7b0/doc/index.rst]: @nordicjm fixed

File: boards/weact/mini_stm32h7b0/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+   - USB Type-C Connector
+   - SWD header for external debugger
+   - 2x 40-pin GPIO connector
+
+.. figure:: img/stm32h7xx.webp
+      :align: center
+      :alt: MiniSTM32H7B0 Core Board
+
+      MiniSTM32H7B0 Core Board (Credit: WeAct Studio)
+
```
Comment (kartben) [inline boards/weact/mini_stm32h7b0/doc/index.rst:32]:
```suggestion
```

The image auto included in the "wikipedia" side bar is enough.

Bonus point if you rename the image file to be the same as the board name

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/weact/mini_stm32h7b0/doc/index.rst (hunk 2 above)
Comment (ric96) [reply boards/weact/mini_stm32h7b0/doc/index.rst]: fixed.

File: boards/weact/mini_stm32h7b0/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+Key Features
+
+- STM32 microcontroller in LQFP100 package
+- USB OTG or full-speed device
+- 1 user LED
+- User, boot, and reset push-buttons
+- 32.768 kHz and 25MHz HSE crystal oscillators
+- External NOR Flash memories: 64-Mbit Quad-SPI and 64-Mbit SPI
+- Board connectors:
+   - Camera (8 bit) connector
```
Comment (kartben) [inline boards/weact/mini_stm32h7b0/doc/index.rst:20]:
```suggestion

   - Camera (8 bit) connector
```

**Review by kartben — DISMISSED** [review]
+1 but fixing the remaining bullet list glitch would be nice

**Review by kartben — DISMISSED** [review]

**Review by nordicjm — COMMENTED** [review]
Minor nit.

**Review by nordicjm — DISMISSED** [review]

File: boards/weact/mini_stm32h7b0/Kconfig.defconfig [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+if BOARD_MINI_STM32H7B0
+
+if DISPLAY
+
+config INPUT
+	default y
+
+if LVGL
+
+config LV_COLOR_16_SWAP
```
Comment (kartben) [inline boards/weact/mini_stm32h7b0/Kconfig.defconfig:15]: as per #81092:

```suggestion
configdefault LV_COLOR_16_SWAP
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Comment by kartben** [comment]
pushed a small change to fix CI issue due to changes from #81293

**Review by erwango — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**